new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java

```diff
@@ -448,6 +455,63 @@ public static HelmParams installAndVerifyVoyager(String voyagerNamespace,
     return voyagerHelmParams;
   }
 
+  /**
+   * Install traefik and wait for up to five minutes for the traefik pod to be ready.
+   *
+   * @param traefikNamespace the namespace in which the traefik ingress controller is installed
+   * @param nodeportshttp the web nodeport of traefik
+   * @param nodeportshttps the websecure nodeport of traefik
+   * @return the traefik Helm installation parameters
+   */
+  public static HelmParams installAndVerifyTraefik(String traefikNamespace,
+      int nodeportshttp,
+      int nodeportshttps) {
+    LoggingFacade logger = getLogger();
+    // Helm install parameters
+    HelmParams traefikHelmParams = new HelmParams()
+        .releaseName(TRAEFIK_RELEASE_NAME + "-" + traefikNamespace.substring(3))
+        .namespace(traefikNamespace)
+        .repoUrl(TRAEFIK_REPO_URL)
+        .repoName(TRAEFIK_REPO_NAME)
+        .chartName(TRAEFIK_CHART_NAME);
+
+    // NGINX chart values to override
+    TraefikParams traefikParams = new TraefikParams()
+        .helmParams(traefikHelmParams);
+
+    if (nodeportshttp != 0 && nodeportshttps != 0) {
+      traefikParams
+          .nodePortsHttp(nodeportshttp)
+          .nodePortsHttps(nodeportshttps);
+    }
```

Comment:
what if only nodeportshttp or nodeportshttps is not 0?

Reply:
I copied this code from the one I guess you created. Your question makes sense, removed the check and assign value only from supplied method params.